pkg/virt-handler/vm_test.go

```diff
@@ -794,139 +794,130 @@ var _ = Describe("VirtualMachineInstance", func() {
-			vmi.UID = vmiTestUUID
-			vmi.ObjectMeta.ResourceVersion = "1"
-			vmi.Status.Phase = v1.Running
-			vmi = addActivePods(vmi, podTestUUID, host)
+		Context("access credential synced condition", func() {
+			var (
+				vmi    *v1.VirtualMachineInstance
+				domain *api.Domain
+			)
 
-			domain := api.NewMinimalDomainWithUUID("testvmi", vmiTestUUID)
-			domain.Status.Status = api.Running
-			domain.Spec.Metadata.KubeVirt.AccessCredential = &api.AccessCredentialMetadata{
-				Succeeded: true,
-				Message:   "",
+			BeforeEach(func() {
+				vmi = libvmi.New(
+					libvmi.WithNamespace(k8sv1.NamespaceDefault),
+					libvmi.WithResourceMemory("8192Ki"),
+				)
+
+				vmi.UID = vmiTestUUID
+				vmi.ObjectMeta.ResourceVersion = "1"
+				vmi.Status.Phase = v1.Running
+				vmi = addActivePods(vmi, podTestUUID, host)
+
+				domain = api.NewMinimalDomainWithUUID(vmi.Name, vmiTestUUID)
+				domain.Status.Status = api.Running
+			})
+
+			prepare := func() {
+				vmiFeeder.Add(vmi)
+				domainFeeder.Add(domain)
+				createVMI(vmi)
+
+				client.EXPECT().SyncVirtualMachine(vmi, gomock.Any())
+				mockHotplugVolumeMounter.EXPECT().Unmount(gomock.Any(), mockCgroupManager).Return(nil)
+				mockHotplugVolumeMounter.EXPECT().Mount(gomock.Any(), mockCgroupManager).Return(nil)
 			}
 
-			vmiFeeder.Add(vmi)
-			domainFeeder.Add(domain)
-			createVMI(vmi)
+			It("should add condition when credentials report success", func() {
+				domain.Spec.Metadata.KubeVirt.AccessCredential = &api.AccessCredentialMetadata{
+					Succeeded: true,
+					Message:   "",
+				}
 
-			client.EXPECT().SyncVirtualMachine(vmi, gomock.Any())
-			mockHotplugVolumeMounter.EXPECT().Unmount(gomock.Any(), mockCgroupManager).Return(nil)
-			mockHotplugVolumeMounter.EXPECT().Mount(gomock.Any(), mockCgroupManager).Return(nil)
+				prepare()
 
-			sanityExecute()
+				sanityExecute()
 
-			expectEvent(string(v1.AccessCredentialsSyncSuccess), true)
-			updatedVMI, err := virtfakeClient.KubevirtV1().VirtualMachineInstances(metav1.NamespaceDefault).Get(context.TODO(), vmi.Name, metav1.GetOptions{})
-			Expect(err).NotTo(HaveOccurred())
-			Expect(updatedVMI.Status.Conditions).To(ConsistOf(
-				MatchFields(IgnoreExtras, Fields{
-					"Type":   Equal(v1.VirtualMachineInstanceAccessCredentialsSynchronized),
-					"Status": Equal(k8sv1.ConditionTrue)},
-				),
-				MatchFields(IgnoreExtras, Fields{
-					"Type":   Equal(v1.VirtualMachineInstanceIsMigratable),
-					"Status": Equal(k8sv1.ConditionTrue)},
-				),
-				MatchFields(IgnoreExtras, Fields{
-					"Type":   Equal(v1.VirtualMachineInstanceIsStorageLiveMigratable),
-					"Status": Equal(k8sv1.ConditionTrue)},
-				),
-			))
-		})
+				expectEvent(string(v1.AccessCredentialsSyncSuccess), true)
+				updatedVMI, err := virtfakeClient.KubevirtV1().VirtualMachineInstances(metav1.NamespaceDefault).Get(context.TODO(), vmi.Name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(updatedVMI.Status.Conditions).To(ContainElement(
```

Comment:
You could use `matcher.HaveConditionTrue` instead.

Reply:
done